### ISSUE TITLE
How to set up `GRPCRoute` for any method and service?

### ISSUE THREAD

**Issue by kandeshvari** [projectcontour/contour#5374]
How to set up `GRPCRoute` for any methods and services?

If I omit this section, I get the error:

```
GRPCRoute.Spec.Rules.Matches.Method: Both Service and Method need be configured.
```

My manifest:

```
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: GRPCRoute
metadata:
  name: tempo-grpc
  namespace: tempo
spec:
  hostnames:
    - tempo-otlp.k8s.local
  parentRefs:
    - name: tempo-grpc
      group: gateway.networking.k8s.io
      kind: Gateway
      namespace: tempo
  rules:
    - backendRefs:
        - name: tempo-distributor
          port: 4317
```          

**Environment:**

- Contour version: 1.25
- Kubernetes version: 1.26

**Comment by skriss** [comment]
Hey @kandeshvari, I made a change in upstream Gateway API to allow this (ref. https://github.com/kubernetes-sigs/gateway-api/commit/8acd4fca41b17fe9be49369e41fd4e5e95136744), but it's not available until Gateway API v0.7.0.

You _could_ install Gateway API v0.7.0 into your cluster ([experimental install](https://github.com/kubernetes-sigs/gateway-api/releases/download/v0.7.0/experimental-install.yaml)) to get access to this functionality -- it's not _officially_ supported with Contour v1.25.0 (which uses Gateway API v0.6.2), but it looks like it should be compatible.

**Comment by kandeshvari** [comment]
It works! Thank you for the advice! I simply replaced `GRPCRoute` with a new one from 0.7.0 and routes have been created. Thank you!
